src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtilsUnitTest.java

```diff
@@ -30,40 +34,223 @@
+            testArgs.add(
+                    new Object[]{
+                            DataSourceUtils.MAX_MAJOR_VERSION_NUMBER, minor, DataSourceUtils.MAX_DATE,
+                            (minor == DataSourceUtils.MIN_MINOR_VERSION_NUMBER || minor == DataSourceUtils.MAX_MINOR_VERSION_NUMBER)
+                    }
+            );
+        }
 
-                if ( i < baseArgs.size() - 1) {
-                    argList.addAll(baseArgs.subList(i + 1, baseArgs.size()));
+        // major > MAX_MAJOR (minor shouldn't matter)
+        for (int minor = 0; minor < MINOR_IT_MAX_VAL; ++minor) {
+            testArgs.add(
+                    new Object[]{
+                            DataSourceUtils.MAX_MAJOR_VERSION_NUMBER + 1, minor, DataSourceUtils.MAX_DATE, false
+                    }
+            );
+        }
+
+        // =============================================================
+        // Next do tests that check the dates themselves:
+        // -------------------------------------------------------------
+
+        // ----------------------------------------
+        // Dates in range:
+        // ----------------------------------------
+
+        // Min version with minimally / maximally acceptable dates:
+        testArgs.add(
+                new Object[]{
+                        DataSourceUtils.MIN_MAJOR_VERSION_NUMBER,
+                        DataSourceUtils.MIN_MINOR_VERSION_NUMBER,
+                        DataSourceUtils.MIN_DATE,
+                        true
+                }
+        );
+        testArgs.add(
+                new Object[]{
+                        DataSourceUtils.MIN_MAJOR_VERSION_NUMBER,
+                        DataSourceUtils.MIN_MINOR_VERSION_NUMBER,
+                        DataSourceUtils.MAX_DATE,
+                        true
                 }
+        );
 
-                argList.add(offset >= 0);
+        // Max version with minimally / maximally acceptable dates:
+        testArgs.add(
+                new Object[]{
+                        DataSourceUtils.MAX_MAJOR_VERSION_NUMBER,
+                        DataSourceUtils.MAX_MINOR_VERSION_NUMBER,
+                        DataSourceUtils.MIN_DATE,
+                        true
+                }
+        );
+        testArgs.add(
+                new Object[]{
+                        DataSourceUtils.MAX_MAJOR_VERSION_NUMBER,
+                        DataSourceUtils.MAX_MINOR_VERSION_NUMBER,
+                        DataSourceUtils.MAX_DATE,
+                        true
+                }
+        );
 
-                testArgs.add(argList.toArray());
-            }
+        // Year / Month inside acceptable dates with any valid day:
+        for (int day = 1; day < 30; ++day) {
+            testArgs.add(
+                    new Object[]{
+                            DataSourceUtils.MIN_MAJOR_VERSION_NUMBER,
+                            DataSourceUtils.MIN_MINOR_VERSION_NUMBER,
+                            LocalDate.of(2020, 4, day),
```

Comment:
This should be relative to one of the MIN/MAX constants, otherwise it will break in the future.

Reply:
Yeah - I didn't want to have to deal with month / year rollover when doing math.  I'll fix it now.